### ISSUE TITLE
Data race in limiter.go 

### ISSUE THREAD

**Issue by fxfactorial** [libp2p/go-libp2p#902]
Not sure if I already posted this, but this function `.addCheckFdLimit` seems new to me

```
WARNING: DATA RACE
Read at 0x00c000c5a5e0 by goroutine 715:
  github.com/gorilla/websocket.(*messageWriter).flushFrame()
      /home/edgar/go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:614 +0x7b5
  github.com/gorilla/websocket.(*messageWriter).Close()
      /home/edgar/go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:724 +0xba
  github.com/gorilla/websocket.(*Conn).WriteMessage()
      /home/edgar/go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:773 +0x315
  github.com/libp2p/go-ws-transport.(*Conn).Write()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-ws-transport@v0.3.0/conn_native.go:70 +0x92
  go.(*struct { net.Conn; github.com/multiformats/go-multiaddr-net.maEndpoints }).Write()
      <autogenerated>:1 +0x87
  crypto/tls.(*Conn).write()
      /home/edgar/.gimme/versions/go1.14.1.linux.amd64/src/crypto/tls/conn.go:895 +0x243
  crypto/tls.(*Conn).writeRecordLocked()
      /home/edgar/.gimme/versions/go1.14.1.linux.amd64/src/crypto/tls/conn.go:944 +0x56d
  crypto/tls.(*Conn).sendAlertLocked()
      /home/edgar/.gimme/versions/go1.14.1.linux.amd64/src/crypto/tls/conn.go:796 +0xac
  crypto/tls.(*Conn).closeNotify()
      /home/edgar/.gimme/versions/go1.14.1.linux.amd64/src/crypto/tls/conn.go:1315 +0x133
  crypto/tls.(*Conn).Close()
      /home/edgar/.gimme/versions/go1.14.1.linux.amd64/src/crypto/tls/conn.go:1288 +0x140
  github.com/libp2p/go-libp2p-tls.(*conn).Close()
      <autogenerated>:1 +0x5d
  github.com/libp2p/go-libp2p-transport-upgrader.(*Upgrader).setupMuxer()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.2.0/upgrader.go:128 +0x266
  github.com/libp2p/go-libp2p-transport-upgrader.(*Upgrader).upgrade()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.2.0/upgrader.go:92 +0x581
  github.com/libp2p/go-libp2p-transport-upgrader.(*Upgrader).UpgradeOutbound()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.2.0/upgrader.go:58 +0x125
  github.com/libp2p/go-ws-transport.(*WebsocketTransport).Dial()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-ws-transport@v0.3.0/websocket.go:67 +0x177
  github.com/libp2p/go-libp2p-swarm.(*Swarm).dialAddr()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/swarm_dial.go:516 +0x270
  github.com/libp2p/go-libp2p-swarm.(*Swarm).dialAddr-fm()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/swarm_dial.go:504 +0x96
  github.com/libp2p/go-libp2p-swarm.(*dialLimiter).executeDial()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/limiter.go:218 +0x29f

Previous write at 0x00c000c5a5e0 by goroutine 866:
  github.com/gorilla/websocket.(*Conn).SetWriteDeadline()
      /home/edgar/go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:781 +0xd0
  github.com/libp2p/go-ws-transport.(*Conn).SetWriteDeadline()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-ws-transport@v0.3.0/conn_native.go:120 +0x9e
  github.com/libp2p/go-ws-transport.(*Conn).SetDeadline()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-ws-transport@v0.3.0/conn_native.go:112 +0xfa
  go.(*struct { net.Conn; github.com/multiformats/go-multiaddr-net.maEndpoints }).SetDeadline()
      <autogenerated>:1 +0x84
  crypto/tls.(*Conn).SetDeadline()
      /home/edgar/.gimme/versions/go1.14.1.linux.amd64/src/crypto/tls/conn.go:131 +0x72
  github.com/libp2p/go-libp2p-tls.(*conn).SetDeadline()
      <autogenerated>:1 +0x7f
  github.com/libp2p/go-stream-muxer-multistream.(*Transport).NewConn()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-stream-muxer-multistream@v0.2.0/multistream.go:64 +0x329
  github.com/libp2p/go-libp2p-transport-upgrader.(*Upgrader).setupMuxer.func1()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.2.0/upgrader.go:120 +0xd4

Goroutine 715 (running) created at:
  github.com/libp2p/go-libp2p-swarm.(*dialLimiter).addCheckFdLimit()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/limiter.go:168 +0x2dc
  github.com/libp2p/go-libp2p-swarm.(*dialLimiter).addCheckPeerLimit()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/limiter.go:182 +0x791
  github.com/libp2p/go-libp2p-swarm.(*dialLimiter).AddDialJob()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/limiter.go:193 +0x169
  github.com/libp2p/go-libp2p-swarm.(*Swarm).limitedDial()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/swarm_dial.go:496 +0xc08
  github.com/libp2p/go-libp2p-swarm.(*Swarm).dialAddrs()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/swarm_dial.go:462 +0xad5
  github.com/libp2p/go-libp2p-swarm.(*Swarm).dial()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/swarm_dial.go:362 +0x6f1
  github.com/libp2p/go-libp2p-swarm.(*Swarm).doDial()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/swarm_dial.go:290 +0x211
  github.com/libp2p/go-libp2p-swarm.(*Swarm).doDial-fm()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/swarm_dial.go:275 +0x70
  github.com/libp2p/go-libp2p-swarm.(*activeDial).start()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.2.3/dial_sync.go:84 +0xc3
==================
==================

```

**Comment by Stebalien** [comment]
Ah, we hadn't updated the dep in go-libp2p. Fixing that now.

**Comment by Stebalien** [comment]
Fixed in go-libp2p v0.8.3.
